modules/util/portal-tools-rest-builder/src/test/resources/com/liferay/portal/tools/rest/builder/dependencies/expected/sample-impl/src/main/java/com/example/sample/internal/resource/v1_0_0/BasePolymorphicSchemaResourceImpl.java

```diff
@@ -0,0 +1,610 @@
+import com.liferay.portal.odata.filter.FilterParser;
+import com.liferay.portal.odata.filter.FilterParserProvider;
+import com.liferay.portal.odata.sort.SortField;
+import com.liferay.portal.odata.sort.SortParser;
+import com.liferay.portal.odata.sort.SortParserProvider;
+import com.liferay.portal.vulcan.accept.language.AcceptLanguage;
+import com.liferay.portal.vulcan.batch.engine.VulcanBatchEngineTaskItemDelegate;
+import com.liferay.portal.vulcan.batch.engine.resource.VulcanBatchEngineImportTaskResource;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+import com.liferay.portal.vulcan.resource.EntityModelResource;
+import com.liferay.portal.vulcan.util.ActionUtil;
+import com.liferay.portal.vulcan.util.TransformUtil;
+
+import java.io.Serializable;
+
+import java.lang.reflect.Array;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Generated;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import javax.ws.rs.NotSupportedException;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * @author John Doe
+ * @generated
+ */
+@Generated("")
+@javax.ws.rs.Path("/1.0.0")
+public abstract class BasePolymorphicSchemaResourceImpl
+	implements EntityModelResource, PolymorphicSchemaResource,
+			   VulcanBatchEngineTaskItemDelegate<PolymorphicSchema> {
+
+	/**
+	 * Invoke this method with the command line:
+	 *
+	 * curl -X 'GET' 'http://localhost:8080/o/sample/1.0.0/polymorphic-schemas'  -u 'test@liferay.com:test'
+	 */
+	@io.swagger.v3.oas.annotations.tags.Tags(
+		value = {
+			@io.swagger.v3.oas.annotations.tags.Tag(name = "PolymorphicSchema")
+		}
+	)
+	@javax.ws.rs.GET
+	@javax.ws.rs.Path("/polymorphic-schemas")
+	@javax.ws.rs.Produces({"application/json", "application/xml"})
+	@Override
+	public Page<PolymorphicSchema> getPolymorphicSchemasPage()
+		throws Exception {
+
+		return Page.of(Collections.emptyList());
+	}
+
+	/**
+	 * Invoke this method with the command line:
+	 *
+	 * curl -X 'POST' 'http://localhost:8080/o/sample/1.0.0/polymorphic-schemas' -d $'{"name": ___, "type": ___}' --header 'Content-Type: application/json' -u 'test@liferay.com:test'
+	 */
+	@io.swagger.v3.oas.annotations.tags.Tags(
+		value = {
+			@io.swagger.v3.oas.annotations.tags.Tag(name = "PolymorphicSchema")
+		}
+	)
+	@javax.ws.rs.Consumes({"application/json", "application/xml"})
+	@javax.ws.rs.Path("/polymorphic-schemas")
+	@javax.ws.rs.POST
+	@javax.ws.rs.Produces({"application/json", "application/xml"})
+	@Override
+	public PolymorphicSchema postPolymorphicSchema(
+			PolymorphicSchema polymorphicSchema)
+		throws Exception {
+
+		return new PolymorphicSchema();
```

Comment:
@albertojml this does not compile. `PolymorphicSchema` is abstract, it cannot be instantiated. Maybe we should return the first child instance? null?

Reply:
Also the client module and the test modules do not take into acccount the inheritance.